### ISSUE TITLE
Case matters when trying to auth against sasl?

### ISSUE THREAD

**Issue by wolfsage** [cyrusimap/cyrus-imapd#2257]
If I create a user like this:

     echo 'mypassword' | saslpasswd2 -p -c USER

And test it:

    testsaslauthd -u USER -p mypassword
    0: OK "Success."

Then create its mailbox:

    echo 'createmailbox user/USER@localhost' | /usr/cyrus/bin/cyradm -u imapuser -w secret localhost
    localhost> localhost>

And try to use it:

    curl -u USER@localhost:mypassword -i -X POST 'http://localhost:80/jmap/' -H "Content-Type: application/json" -d '{"using":["ietf:jmapmail"],"methodCalls":[["Email/query", {}, "a"]]}'
    curl: (52) Empty reply from server

````
sudo tail -n 3 /var/log/syslog
Feb 22 10:23:58 cyrus-dev cyrus/http[6759]: badlogin: localhost [::1] Basic USER@localhost SASL(-13): authentication failure: checkpass failed
Feb 22 10:23:58 cyrus-dev kernel: [ 2954.600744] traps: httpd[6759] trap divide error ip:7fb4d5d84eca sp:7ffd7a4ba9a0 error:0 in libdigestmd5.so.2.0.25[7fb4d5d81000+d000]
Feb 22 10:23:58 cyrus-dev cyrus/master[1522]: process type:SERVICE name:http path:/usr/cyrus/libexec/httpd age:0.098s pid:6759 signaled to death by signal 8 (Floating point exception, core dumped)
````

But if I do all of the above replacing `USER` with something like `loweruser` it works.

I think this is a bug - we allow creating a mailbox for `USER@localhost` but then can't make requests against it? (curl -u user@localhost doesn't work)

Note that if I create my user as `myuser` I can make requests like `curl -u myUSer...` and that works

**Comment by hagedose** [comment]
Not sure if this is relevant, but there is a setting in imapd.conf regarding this:

```
       normalizeuid: 0
            Lowercase uid and strip leading and trailing blanks. It is recommended to set this to yes, espe-
            cially if OpenLDAP is used as authentication source.

```

**Comment by hagedose** [comment]
I just realized that the option I mentioned is from a patch by Red Hat. It's not part of the official distribution.

**Comment by brong** [comment]
Yeah, we use tolower in prod at Fastmail.  That's how to avoid this if you want case not to be treated significant
